lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -36,35 +48,60 @@ def execute(self):
-                            p.read_text() for p in self.tools(path)
-                        ]
-                    else:
-                        host_tools[group][path.name] = [p for p in self.tools(path)]
+                try:
+                    for path in self.gen_tools_group_dir(group).glob("*/**"):
+                        host = path.name
+                        if self.context.with_option:
+                            host_tools[group][host] = [
+                                (p, (path / p).read_text().rstrip("\n"))
+                                for p in self.tools(path)
+                            ]
+                        else:
+                            host_tools[group][host] = [p for p in self.tools(path)]
+                except BadToolGroup:
+                    self.logger.error("Tool group does not exist: %s", group)
+                    return 1
             if host_tools:
-                for k, v in host_tools.items():
-                    for h, t in v.items():
-                        print("%s: %s %s" % (k, h, t))
+                self.print_results(host_tools, self.context.with_option)
         else:
             # List the groups which include this tool
             group_list = []
+            tool = self.context.name
+            options = {}
             for group in groups:
-                tg_dir = self.gen_tools_group_dir(group)
-                if not tg_dir.exists():
-                    self.logger.error("bad or missing tool group %s", group)
-                    continue
+                try:
+                    tg_dir = self.gen_tools_group_dir(group)
+                except BadToolGroup:
+                    self.logger.error("Tool group does not exist: %s", group)
+                    return 1
 
                 for path in tg_dir.iterdir():
+                    # skip files like __label__ and __trigger__
+                    if not path.is_dir():
+                        continue
+
+                    host = path.name
                     # Check to see if the tool is in any of the hosts.
-                    if self.context.name in self.tools(path):
+                    if tool in self.tools(path):
                         group_list.append(group)
+                        if self.context.with_option:
+                            options[group] = (
+                                host,
+                                (path / tool).read_text().rstrip("\n"),
+                            )
+
             if group_list:
-                print(
-                    "tool name: %s groups: %s"
-                    % (self.context.name, ", ".join(group_list))
-                )
+                if self.context.with_option:
+                    print(f"tool name: {tool}")
+                    for group, rest in sorted(options.items()):
+                        host, options = rest
+                        print(f"group: {group}, host: {host}, options: {options}")
```

Comment:
My brain is currently fried from PerfConf (and a virus), but Dave's comments now make me question this code:  at line 96, are we supposed to be referencing `options` or `group_list`?  Given that it's `group_list` that we check at line 93 and print at line 100, it seems weird that we would ignore it here (and, it's even stranger that we would overwrite it at line 97...).